setup.py

```diff
@@ -1,109 +1,125 @@
+
+    elif sys.platform.startswith("linux"):
+        modules = []
+        if "--without-libusb" in sys.argv:
+            sys.argv.remove("--without-libusb")
+            hidraw_module = "hid"
         else:
-            src.append(hidapi_src("libusb"))
+            hidraw_module = "hidraw"
+            modules.append(
+                Extension(
+                    "hid",
+                    sources=["hid.pyx", hidapi_src("libusb")],
+                    include_dirs=[embedded_hidapi_include],
+                    extra_compile_args=[pkgconfig.cflags('libusb-1.0 >= 1.0.9')],
+                    extra_link_args=[pkgconfig.libs('libusb-1.0 >= 1.0.9')]
+                )
+            )
         modules.append(
             Extension(
-                "hid",
-                sources=src,
-                include_dirs=[hidapi_include, "/usr/include/libusb-1.0"],
-                libraries=libs,
+                hidraw_module,
+                sources=["hidraw.pyx", hidapi_src("linux")],
+                include_dirs=[embedded_hidapi_include],
+                libraries=["udev"],
             )
         )
-    libs = ["udev", "rt"]
-    src = ["hidraw.pyx"]
-    if system_hidapi == 1:
-        libs.append("hidapi-hidraw")
+
     else:
-        src.append(hidapi_src("linux"))
-    modules.append(
-        Extension(
-            hidraw_module, sources=src, include_dirs=[hidapi_include], libraries=libs,
+        modules = [
+            Extension(
+                "hid",
+                sources=["hid.pyx", hidapi_src("libusb")],
+                include_dirs=[embedded_hidapi_include],
+                extra_compile_args=[pkgconfig.cflags('libusb-1.0 >= 1.0.9')],
+                extra_link_args=[pkgconfig.libs('libusb-1.0 >= 1.0.9')]
+            )
+        ]
+
+    return modules
+
+
+def hid_from_system_hidapi():
+    if sys.platform.startswith("linux"):
+        modules = []
+        if "--without-libusb" in sys.argv:
+            sys.argv.remove("--without-libusb")
+            hidraw_module = "hid"
+        else:
+            hidraw_module = "hidraw"
+            modules.append(
+                Extension(
+                    "hid",
+                    sources=["hid.pyx"],
+                    extra_compile_args=[pkgconfig.cflags('hidapi-libusb >= 0.14')],
+                    extra_link_args=[pkgconfig.libs('hidapi-libusb >= 0.14')]
+                )
+            )
+        modules.append(
+            Extension(
+                hidraw_module,
+                sources=["hidraw.pyx"],
+                extra_compile_args=[pkgconfig.cflags('hidapi-hidraw >= 0.14')],
+                extra_link_args=[pkgconfig.libs('hidapi-hidraw >= 0.14')]
+            )
         )
-    )
-
-if sys.platform.startswith("darwin"):
-    macos_sdk_path = (
-        subprocess.check_output(["xcrun", "--show-sdk-path"]).decode().strip()
-    )
-    os.environ["CFLAGS"] = (
-        '-isysroot "%s" -framework IOKit -framework CoreFoundation -framework AppKit'
```

Comment:
removed `-framework` commands from `CFLAGS` (moved to linker flags) as otherwise I get warnings like:
```sh
clang: warning: -framework IOKit: 'linker' input unused [-Wunused-command-line-argument]
```